main.go

```diff
@@ -441,33 +446,50 @@ func handleCliVersion(cCtx *cli.Context) error {
 	return nil
 }
 
+// must fetch and generate an SDK. Must return an error if generation failed or nothing is generated
 func fetchManifestAndGenerateSDK(c *ContractConfig, host string) error {
 	m, err := fetchManifest(&c.ScriptHash, host)
 	if err != nil {
 		return err
 	}
 
-	languages := cfg.Defaults.OnChain.Languages
+	var onChainLanguages []string = nil
 	if c.OnChain != nil {
-		languages = c.OnChain.Languages
+		onChainLanguages = c.OnChain.Languages
+	} else if cfg.Defaults.OnChain != nil {
+		onChainLanguages = cfg.Defaults.OnChain.Languages
 	}
-	for _, l := range languages {
-		err = generateSDK(&generators.GenerateCfg{Manifest: m, ContractHash: c.ScriptHash, SdkDestination: cfg.getSdkDestination(l)}, l, SDK_ONCHAIN)
-		if err != nil {
-			return err
+
+	if onChainLanguages != nil {
+		for _, l := range onChainLanguages {
+			err = generateSDK(&generators.GenerateCfg{Manifest: m, ContractHash: c.ScriptHash, SdkDestination: cfg.getSdkDestination(l, generators.SDK_ONCHAIN)}, l, generators.SDK_ONCHAIN)
+			if err != nil {
+				return err
+			}
 		}
 	}
 
-	languages = cfg.Defaults.OffChain.Languages
+	var offChainLanguages []string = nil
 	if c.OffChain != nil {
-		languages = c.OffChain.Languages
+		offChainLanguages = c.OffChain.Languages
+	} else if cfg.Defaults.OffChain != nil {
+		offChainLanguages = cfg.Defaults.OffChain.Languages
 	}
-	for _, l := range languages {
-		err = generateSDK(&generators.GenerateCfg{Manifest: m, ContractHash: c.ScriptHash, SdkDestination: cfg.getSdkDestination(l)}, l, SDK_OFFCHAIN)
-		if err != nil {
-			return err
+
+	if offChainLanguages != nil {
+		for _, l := range offChainLanguages {
+			err = generateSDK(&generators.GenerateCfg{Manifest: m, ContractHash: c.ScriptHash, SdkDestination: cfg.getSdkDestination(l, generators.SDK_OFFCHAIN)}, l, generators.SDK_OFFCHAIN)
+			if err != nil {
+				return err
+			}
 		}
 	}
+
+	if onChainLanguages == nil && offChainLanguages == nil {
+		return fmt.Errorf("nothing to generate. Ensure your 'cpm.yaml' has an 'onchain' or 'offchain' key under " +
+			"the 'defaults' section or contract specific section with at least one language specified")
+	}
```

Comment:
I've added this check to ensure that a `cpm run` where both the `default` section and contract specific section in `cpm.yaml`  are empty generates an error because returning `nil` indicates "generation success".